### PR TITLE
Add a score threshold search parameter

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue
Fixes https://github.com/meilisearch/meilisearch/issues/4609

## What does this PR do?
- See [usage](https://meilisearch.notion.site/Filter-by-score-usage-224a183ce7b24ca99b6a9a8da755668a?pvs=25#95b76ded400342ba9ab3d67c734836f0) and [the known limitation](https://meilisearch.notion.site/Filter-by-score-usage-224a183ce7b24ca99b6a9a8da755668a?pvs=25#e4e32195bf0e4195b5daecdbb7a97a17)

### CODE REVIEW

File: meilisearch/src/routes/indexes/search.rs [hunk 1]
```diff
@@ -82,6 +83,21 @@ pub struct SearchQueryGet {
     pub hybrid_embedder: Option<String>,
     #[deserr(default, error = DeserrQueryParamError<InvalidSearchSemanticRatio>)]
     pub hybrid_semantic_ratio: Option<SemanticRatioGet>,
+    #[deserr(default, error = DeserrQueryParamError<InvalidSearchRankingScoreThreshold>, default)]
```
Comment (dureuill) [inline meilisearch/src/routes/indexes/search.rs:86]: remove one default

```suggestion
    #[deserr(default, error = DeserrQueryParamError<InvalidSearchRankingScoreThreshold>)]
```

File: meilisearch/src/search.rs [hunk 2]
```diff
@@ -661,6 +709,7 @@ fn prepare_search<'t>(
 ) -> Result<(milli::Search<'t>, bool, usize, usize), MeilisearchHttpError> {
     let mut search = index.search(rtxn);
     search.time_budget(time_budget);
+    search.ranking_score_threshold(query.ranking_score_threshold.map(|rst| rst.0));
```
Comment (dureuill) [inline meilisearch/src/search.rs:712]: check this

File: milli/src/search/mod.rs [hunk 3]
```diff
@@ -146,6 +148,14 @@ impl<'a> Search<'a> {
         self
     }
 
+    pub fn ranking_score_threshold(
+        &mut self,
+        ranking_score_threshold: Option<f64>,
```
Comment (dureuill) [inline milli/src/search/mod.rs:153]:
```suggestion
        ranking_score_threshold: f64,
```

File: milli/src/search/mod.rs [hunk 4]
```diff
@@ -146,6 +148,14 @@ impl<'a> Search<'a> {
...
+    ) -> &mut Search<'a> {
+        self.ranking_score_threshold = ranking_score_threshold;
```
Comment (dureuill) [inline milli/src/search/mod.rs:155]:
```suggestion
        self.ranking_score_threshold = Some(ranking_score_threshold);
```

File: milli/src/search/new/bucket_sort.rs [hunk 5]
```diff
@@ -164,7 +166,9 @@ pub fn bucket_sort<'ctx, Q: RankingRuleQueryTrait>(
             loop {
                 let bucket = std::mem::take(&mut ranking_rule_universes[cur_ranking_rule_index]);
                 ranking_rule_scores.push(ScoreDetails::Skipped);
+
```
Comment (dureuill) [inline milli/src/search/new/bucket_sort.rs:169]: move added logic from maybe_add_to_results

**Review by ManyTheFish — APPROVED** [review]
bors merge

**Pull request merged**